test/requests/charging-module/calculate-charge.request.test.js

```diff
@@ -0,0 +1,157 @@
+
+// Thing under test
+const CalculateChargeRequest = require('../../../app/requests/charging-module/calculate-charge.request.js')
+
+describe('Charging Module Calculate Charge request', () => {
+  const transactionData = _transactionData()
+
+  afterEach(() => {
+    Sinon.restore()
+  })
+
+  describe('when the charge can be calculated', () => {
+    beforeEach(async () => {
+      Sinon.stub(ChargingModuleRequest, 'post').resolves({
+        succeeded: true,
+        response: {
+          info: {
+            gitCommit: '273604040a47e0977b0579a0fef0f09726d95e39',
+            dockerTag: 'ghcr.io/defra/sroc-charging-module-api:v0.19.0'
+          },
+          statusCode: 200,
+          body: {
+            calculation: {
+              chargeValue: 7000,
+              baseCharge: 9700,
+              waterCompanyChargeValue: 800,
+              supportedSourceValue: 3500,
+              winterOnlyFactor: null,
+              section130Factor: null,
+              section127Factor: 0.5,
+              compensationChargePercent: null
+            }
+          }
+        }
+      })
+    })
+
+    it('returns a `true` success status', async () => {
+      const result = await CalculateChargeRequest.send(transactionData)
+
+      expect(result.succeeded).to.be.true()
+    })
+
+    it('returns the results of the calculation in the `response`', async () => {
+      const result = await CalculateChargeRequest.send(transactionData)
+
+      expect(result.response.body.calculation.chargeValue).to.equal(7000)
+      expect(result.response.body.calculation.baseCharge).to.equal(9700)
+      expect(result.response.body.calculation.waterCompanyChargeValue).to.equal(800)
+      expect(result.response.body.calculation.supportedSourceValue).to.equal(3500)
+      expect(result.response.body.calculation.winterOnlyFactor).to.be.null()
+      expect(result.response.body.calculation.section130Factor).to.be.null()
+      expect(result.response.body.calculation.section127Factor).to.equal(0.5)
+      expect(result.response.body.calculation.compensationChargePercent).to.be.null()
+    })
+  })
+
+  describe('when the request cannot calculate a charge', () => {
+    describe('because the request did not return a 2xx/3xx response', () => {
+      beforeEach(async () => {
+        Sinon.stub(ChargingModuleRequest, 'post').resolves({
+          succeeded: false,
+          response: {
+            info: {
+              gitCommit: '273604040a47e0977b0579a0fef0f09726d95e39',
+              dockerTag: 'ghcr.io/defra/sroc-charging-module-api:v0.19.0'
+            },
+            statusCode: 401,
+            body: {
+              statusCode: 401,
+              error: 'Unauthorized',
+              message: 'Invalid JWT: Token format not valid',
+              attributes: { error: 'Invalid JWT: Token format not valid' }
+            }
+          }
+        })
+      })
+
+      it('returns a `false` success status', async () => {
```

Comment:
I don't think we use backticks for this. In other unit tests, we do double quotation marks and then singles inside the sentence.

`it("returns a 'false' success status", async () => {`

Reply:
We have done this in the past, what I have there is a copy and paste from another test. Have a look at this one that @Cruikshanks did `test/requests/charging-module/create-customer-change.request.test.js`. Will check in our dev catch-up this morning to see if we have a standard for doing this kinda thing.